### PR TITLE
#3243 changed response code for page-not-found to 404

### PR DESCRIPTION
**Pull request by phoenix-bjoern**
### Related issues
#3243 

closes #

### Short description and why it's useful
HTTP status code is now 404 instead of a 302 redirect. Note that because of the client side rendering the routing was changed in the theme you catch the 404 returned by the SSR.

### CODE REVIEW

File: core/pages/PageNotFound.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ export default {
   async asyncData ({ store, route, context }) { // this is for SSR purposes to prefetch data
     Logger.log('Entering asyncData for PageNotFound ' + new Date())()
     if (context) context.output.cacheTags.add(`page-not-found`)
+    if (context) context.server.response.statusCode = 404
```
Comment (lukeromanowicz) [inline core/pages/PageNotFound.js]: there is no need for a second if that is the same as above

**Comment by lukeromanowicz** [comment]
Nice job! If it's not a problem, please add one suggested code change and we will be ready for QA

**Comment by pkarw** [comment]
OK, we need some QA in here @GabiDivante @alinadivante @ArturDivante

**Comment by ArturDivante** [comment]
> OK, we need some QA in here @GabiDivante @alinadivante @ArturDivante

I'll look into it today :)

File: docs/guide/upgrade-notes/README.md [hunk 2]
```diff
@@ -73,6 +73,8 @@ core/modules/wishlist/store/actions.ts
 ```
 If by some reasons you wan't to have the `localStorage` back on for `Products by SKU`, `Url Routes` and `SyncTasks` - please juset set these variables back to `false` in your `config/local.json`.
 
+- New page-not-found handling requires to update router/index.js in the theme.
```
Comment (patzick) [inline docs/guide/upgrade-notes/README.md]:
```suggestion
- New page-not-found handling requires to update router/index.js in the theme. Change path of 'page-not-found' route to '*'
```

**Comment by ArturDivante** [comment]
Looks ok.  404 appears in console, URLis not changed.

**Pull request merged**